lib/registration.pm

```diff
@@ -159,7 +159,7 @@ sub fill_in_registration_data {
                 # no need to input registration code if register via SMT
                 last if (get_var('SMT_URL'));
                 $uc_addon = uc $addon;    # change to uppercase to match variable
-                if (my $regcode = get_var("SCC_REGCODE_$uc_addon")) {
+                if (my $regcode = get_var("SCC_REGCODE_$uc_addon", get_required_var('SCC_REGCODE'))) {
```

Comment:
what point has the if then? I think the whole point here is that some modules simply do not require a regcode and so there is no point in typing one. You now just type the same regcode all the time - but this seems to be just a yast/scc bug about the packagehub. If it does not require a regcode, then why does it require the user to type one.